compiler/vectorized_test.go

```diff
@@ -24,50 +40,56 @@ func TestVectorizedFns(t *testing.T) {
 		name         string
 		fn           string
 		vectorizable bool
+		allocated    int64
+		maxAllocated int64
 		inType       semantic.MonoType
-		input        values.Object
-		want         values.Value
+		input        map[string]interface{}
+		want         map[string]interface{}
 		skipComp     bool
 	}{
 		{
 			name:         "field access",
 			fn:           `(r) => ({c: r.a, d: r.b})`,
 			vectorizable: true,
+			allocated:    256, // This should be 128 (2 vectorized fields, each containing 1 int64)
+			maxAllocated: 448,
```

Comment:
> This should be 128

I don't get what you're saying with this comment.  Can you explain? 

The numbers here are going to be dependent on the internals of Arrow arrays, so I'm not sure we want to get specific about byte counts here.

Can we write meaningful tests if we just:
- Allocate memory for the input
- Store the current number of bytes allocated
- Run the function, and collect the output
- Verify that the new current bytes allocated is what we expect; either the same as before, or possibly more if we have created new fields on the output record. Which one it is will depend on the test case I guess. For the time being, it should always be the same. Once we can handle things like `(r) => (r with a: r.a, b: r.b, ab: r.a + r.b)}` it will be more.

I would really like to be able to capture the reference counts on the arrays themselves.

@jsternberg any thoughts on this?

Reply:
I think probably just running it and making sure the numbers make sense or just using them as-is is probably good enough. Some of these numbers can get weird because arrow allocates a minimum amount regardless of how big the array is. I forget the minimum though.

Reply:
That makes sense. In light of this I think this is good as-is. We can always tweak these tests later.